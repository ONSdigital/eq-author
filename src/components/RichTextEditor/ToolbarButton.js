import PropTypes from "prop-types";
import styled, { css } from "styled-components";

import { colors } from "constants/theme";

const activeState = css`
  opacity: 1 !important;
  background-color: ${colors.blue};
  svg * {
    fill: ${colors.white};
  }
`;

const ToolbarButton = styled.button.attrs({ type: "button" })`
  display: block;
  opacity: 0.7;
  padding: 0;
  border: none;
  font-size: 1rem;
  margin: 0;

  svg {
    vertical-align: middle;
  }

  &:hover {
    background-color: ${colors.blue};
    outline: none;

    svg * {
      fill: ${colors.white};
    }
  }

  &:focus {
    outline: 3px solid ${colors.highlight};
    outline-offset: -3px;
  }

  &[disabled] {
    opacity: 0.2;
    pointer-events: none;
  }

  ${props => props.active && activeState};
`;

ToolbarButton.propTypes = {
  active: PropTypes.bool
};

export default ToolbarButton;
