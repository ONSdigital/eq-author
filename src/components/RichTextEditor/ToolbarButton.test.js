import React from "react";

import ToolbarButton from "components/RichTextEditor/ToolbarButton";

import { mount } from "enzyme";

describe("ToolbarButton", () => {
  it("matches snapshot", () => {
    const wrapper = mount(<ToolbarButton>Button</ToolbarButton>); // eslint-disable-line react/jsx-no-bind
    expect(wrapper).toMatchSnapshot();
  });

  it("should style appropriately when active", () => {
    const wrapper = mount(<ToolbarButton active>Button</ToolbarButton>); // eslint-disable-line react/jsx-no-bind
    expect(wrapper).toMatchSnapshot();
  });
});
