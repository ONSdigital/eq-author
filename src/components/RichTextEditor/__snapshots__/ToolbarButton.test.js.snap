// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarButton matches snapshot 1`] = `
.c0 {
  display: block;
  opacity: 0.7;
  padding: 0;
  border: none;
  font-size: 1rem;
  margin: 0;
}

.c0 svg {
  vertical-align: middle;
}

.c0:hover {
  background-color: #3B7A9E;
  outline: none;
}

.c0:hover svg * {
  fill: #FFFFFF;
}

.c0:focus {
  outline: 3px solid #008ad2;
  outline-offset: -3px;
}

.c0[disabled] {
  opacity: 0.2;
  pointer-events: none;
}

<ToolbarButton>
  <button
    className="c0"
    type="button"
  >
    Button
  </button>
</ToolbarButton>
`;

exports[`ToolbarButton should style appropriately when active 1`] = `
.c0 {
  display: block;
  opacity: 0.7;
  padding: 0;
  border: none;
  font-size: 1rem;
  margin: 0;
  opacity: 1 !important;
  background-color: #3B7A9E;
}

.c0 svg {
  vertical-align: middle;
}

.c0:hover {
  background-color: #3B7A9E;
  outline: none;
}

.c0:hover svg * {
  fill: #FFFFFF;
}

.c0:focus {
  outline: 3px solid #008ad2;
  outline-offset: -3px;
}

.c0[disabled] {
  opacity: 0.2;
  pointer-events: none;
}

.c0 svg * {
  fill: #FFFFFF;
}

<ToolbarButton
  active={true}
>
  <button
    className="c0"
    type="button"
  >
    Button
  </button>
</ToolbarButton>
`;
